src/index.js

```diff
@@ -164,6 +164,17 @@ class Branch {
       console.warn("setDMAParamsForEEA: Unable to set DMA params.");
     }
   };
+
+  /*** Set the Branch API's base URL (eg. "https://api2.branch.io") ***/
+  setAPIUrl = (apiUrl) => {
```

Comment:
I recommend removing the JavaScript API and moving this to the native JSON api. The JS <> native runtime switch will be a race condition against this setting which must be called before session init. 

Instead, the json can be read on the native init's start up.